packages/editor/src/components/editor-canvas/edit-template-blocks-notification.js

```diff
@@ -58,18 +49,17 @@ export default function EditTemplateBlocksNotification( { contentRef } ) {
 
 	useEffect( () => {
 		const handleClick = async ( event ) => {
-			if ( renderingMode !== 'template-locked' ) {
-				return;
-			}
```

Comment:
Just double-checking that I'm following correctly: the `renderingMode` is removed from the checks in the click and double click handler because we're moving the check up to where the component is being output instead of making it internal to the component?

Reply:
Yeah you have it right.

It's not necessary for this PR at all—leftover code from some explorations I was doing that went nowhere.

But I kept these changes as I think it's much nicer to simply not mount the component at all if it's not needed rather than unnecessarily mount the component and attach all these unnecessary event listeners that run code on every click.